### PR TITLE
Move featuresAt ref work out of the worker/featureTree, contained in style

### PR DESCRIPTION
**Pull request by lbud**
This essentially reverts #901, moving the featuresAt refs work to be contained within the Style so the worker doesn't parse ref layers into the featureTree.

### CODE REVIEW

File: js/style/style.js [hunk 1]
```diff
@@ -453,6 +457,18 @@ Style.prototype = util.inherit(Evented, {
         }, () => {
             if (error) return callback(error);
 
+            var refFeatures = [];
+            if (Object.keys(this.refs).length) features.forEach((feature) => {
+                if (this.refs[feature.layer.id] && this.refs[feature.layer.id].length) {
+                    this.refs[feature.layer.id].forEach(refLayer => {
+                        var copiedFeature = JSON.parse(JSON.stringify(feature));
```
Comment (lbud) [inline js/style/style.js]: @jfirebaugh is there a less ugly way to deep clone here?

Comment (jfirebaugh) [inline js/style/style.js]: You only need a one level deep clone right? `var copiedFeature = util.extend({}, feature)`.

Comment (lbud) [inline js/style/style.js]: :+1: thanks

**Comment by mourner** [comment]
Merging this now because of the performance regression in #901 (refs created duplicate buckets which increased tile parsing time 2x — water, buildings, etc.). This approach looks much better.

**Pull request merged**